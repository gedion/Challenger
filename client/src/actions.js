export const TOGGLE_CODE_DRAWER = 'TOGGLE_CODE_DRAWER';
export const SET_SELECTED_CODE_DRAWER_ITEM = 'SET_SELECTED_CODE_DRAWER_ITEM';
export const SHOW_APP_INIT = 'SHOW_APP_INIT';
export const TOGGLE_PROGRESS_BAR = 'TOGGLE_PROGRESS_BAR';
import axios from 'axios';

export function toggleCodeDrawer (dispatch) {
  return dispatch({ type: TOGGLE_CODE_DRAWER });
}


export function toggleProgressBar (dispatch) {
  return dispatch({ type: TOGGLE_PROGRESS_BAR });
}

export function setDrawerListItem (dispatch, item) {
    toggleProgressBar(dispatch);
    return axios.get('/api/run/cryptSolution')
    .then((res) => {
      toggleProgressBar(dispatch);
      dispatch({ type: SET_SELECTED_CODE_DRAWER_ITEM,
        data: {
          runResult: res.data ,
          selectedCodeDrawerItem: item
        }
      });
    }).catch(error => {
      toggleProgressBar(dispatch);
      dispatch({ type: SET_SELECTED_CODE_DRAWER_ITEM,
        data: {
          selectedCodeDrawerItem: item
        }
      });
    });
}

