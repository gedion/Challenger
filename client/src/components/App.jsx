import React from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';
import PropTypes from 'prop-types';
import AppBar from 'react-toolbox/lib/app_bar';
import Navigation from 'react-toolbox/lib/navigation';
import Link from 'react-toolbox/lib/link';
import CodeDrawer from './CodeDrawer';
import CodeCard from './CodeCard';
import { Grid, Row, Col } from 'react-flexbox-grid';

import { toggleCodeDrawer, setDrawerListItem}
  from '../actions';

class App extends React.Component {
  constructor (props) {
    super(props);
    this.handleCodeDrawerToggle = this.handleCodeDrawerToggle.bind(this);
    this.handeCodeDrawerListItemClick= this.handeCodeDrawerListItemClick.bind(this);
  }

  handleCodeDrawerToggle () {
    this.props.toggleCodeDrawer();
  }

  handeCodeDrawerListItemClick (event) {
    this.props.setDrawerListItem(_.get(event, 'target.innerText'));
  }

  appBarTest () {
    return (<AppBar title={this.props.title} leftIcon='menu' >
      <Navigation type='horizontal'>
        <Link href='http://' label='Inbox' icon='inbox' />
        <Link href='http://' active label='Profile' icon='person' />
      </Navigation>
    </AppBar>);
  }

  render () {
    return (
      <div>
        {this.appBarTest()}
        <CodeDrawer show={this.props.showCodeDrawer}
          handleToggle={this.handleCodeDrawerToggle}
          onListItemClick={this.handeCodeDrawerListItemClick}
          codeCategories={this.props.codeCategories}
        />
      <Grid fluid>
        <Row>
          {_.map(this.props.selectedCodeCards, ((card, index) => {
            return (<Col xs={6} md={3} key={index} >
              <CodeCard {...card} />
            </Col>);
          }))}
        </Row>
      </Grid>
      </div>
    );
  }
}

App.propTypes = {
  title: PropTypes.string,
  showCodeDrawer: PropTypes.bool,
  toggleCodeDrawer: PropTypes.func,
  codeCategories: PropTypes.array,
  selectedCodeCard: PropTypes.array,
  category: PropTypes.string
};

const mapDispatchToProps = (dispatch, ownProps) => {
  return {
    toggleCodeDrawer: () => {
      toggleCodeDrawer(dispatch);
    },
    setDrawerListItem: (item) => {
      setDrawerListItem(dispatch, item);
    },
    dispatch
  };
};

const mapStateToProps = state => {
  return state;
};

const mergeProps = (stateProps, dispatchProps, ownProps) => {
  return {
    ...ownProps,
    ...stateProps,
    ...dispatchProps
  };
};

App.propTypes = {
};

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(App);
