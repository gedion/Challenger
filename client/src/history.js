import createHistory from 'history/createBrowserHistory';

// We create the history object here so that it is a singleton.
export default createHistory();
