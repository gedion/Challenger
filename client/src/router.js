// Initially taken from
// https://medium.freecodecamp.com/you-might-not-need-react-router-38673620f3d.
// A more complete implementation is at
// https://github.com/kriasoft/universal-router.

import pathtoRegexp from 'path-to-regexp';

// The path parameter is the path specified by the route.
// The uri is the location (presumably, the current location, for which
// we are trying to find a route.
function matchURI (path, uri) {
  const keys = [];
  const regex = pathtoRegexp(path, keys);
  const match = regex.exec(uri);
  if (!match) return null;
  const params = Object.create(null);
  for (let i = 1; i < match.length; i++) {
    // TODO: Why the extra logic for undefined?
    params[keys[i - 1].name] =
                      match[i] !== undefined ? match[i] : undefined;
  }
  return params;
}

async function resolve (routes, context) {
  for (const route of routes) {
    const uri = context.error ? '/error' : context.pathname;

    // params has an overloaded role here.  If params is empty but not null
    // that's a signal that the route matched the uri even though no recognized
    // params were included in the uri.  If params is null, the route did not
    // match the uri.
    const params = matchURI(route.path, uri);
    if (!params) continue;
    const result = await route.action({ ...context, params });
    if (result) return result;
  }
  const error = new Error('Not found');
  error.status = 404;
  throw error;
}

export default { resolve };
