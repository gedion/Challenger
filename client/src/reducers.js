import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import _ from 'lodash';

let defaultState = { };
import { TOGGLE_CODE_DRAWER, SET_SELECTED_CODE_DRAWER_ITEM,   SHOW_APP_INIT } from './actions';
// global appInit
let initState = _.isUndefined(window.appInit)
  ? defaultState : window.appInit;

const app = function (state, action) {
  switch (action.type) {
    case SHOW_APP_INIT:
      return action.data; 
    case TOGGLE_CODE_DRAWER:
      return Object.assign({}, state, { showCodeDrawer: !state.showCodeDrawer });
    case SET_SELECTED_CODE_DRAWER_ITEM:
      let selectedCodeDrawerItem = action.data;
      let selectedCodeCards = _.filter(state.codeCards, { category: selectedCodeDrawerItem });
      return Object.assign({}, state, {
        showCodeDrawer: !state.showCodeDrawer,
        selectedCodeCards, selectedCodeDrawerItem
      });
    default:
      return state;
  }
};

// const store = createStore(app, initState, applyMiddleware(thunk));

// For development only
const store = createStore(app, initState,
  compose(
    applyMiddleware(thunk),
    window.devToolsExtension ? window.devToolsExtension() : f => f
  ),
  applyMiddleware(thunk)
);

export default store;
