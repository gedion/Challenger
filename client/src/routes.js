import React from 'react';
import App from './components/App';

const routes = [
  { path: '/', action: () => <App /> }
];

export default routes;
