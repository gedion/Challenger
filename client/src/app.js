// See https://medium.freecodecamp.com/you-might-not-need-react-router-38673620f3d
// for how history and routing work.

// Import css as global by bypassing CSS Modules (as discussed in
// https://github.com/css-modules/css-modules/pull/65)

import ReactDOM from 'react-dom';
import React from 'react';
import { Provider } from 'react-redux';
import history from './history';
import router from './router';
import routes from './routes';
import store from './reducers';  // TODO: Reorganize store and reducers.
const container = document.getElementById('react-root');

// For example of wiring up redux, see
// https://github.com/reactjs/redux/blob/master/docs/basics/ExampleTodoList.md

function renderComponent (component) {
  const providerForReduxStore = <Provider store={store}>{component}</Provider>;
  ReactDOM.render(providerForReduxStore, container);
}

function render (location) {
  router.resolve(routes, location)
    .then(renderComponent)
    // This catch adds an error property to the location object.
    .catch(error => {
      console.error(error);
      router.resolve(routes, { ...location, error })
      .then(renderComponent);
    });
}

render(history.location);
history.listen(render);

// See longer version of this in
// https://github.com/kriasoft/react-starter-kit/blob/master/src/client.js.
if (module.hot) {
  module.hot.accept();
}
