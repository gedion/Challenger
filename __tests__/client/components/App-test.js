import React from 'react';
import renderer from 'react-test-renderer';
import App from './../../../client/src/components/App';
import store from './../../../client/src/reducers';
import fs from 'fs';
import path from 'path';
import { SHOW_APP_INIT } from './../../../client/src/actions';

let initAppState = fs.readFileSync(path.join(__dirname, '../../../fixtures/init-app-state.json'), 'utf8');
initAppState = JSON.parse(initAppState);

describe('Tests client components', () => {
  beforeEach(() => {
    store.dispatch({ data: initAppState, type: SHOW_APP_INIT });
  });

  it('renders app correctly', () => {
    const tree = renderer
      .create(<App store={store} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
