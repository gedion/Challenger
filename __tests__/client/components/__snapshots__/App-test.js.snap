// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests client components renders app correctly 1`] = `
<div>
  <header
    className=""
    data-react-toolbox="app-bar"
  >
    <div
      className={undefined}
    >
      <button
        className="undefined"
        data-react-toolbox="button"
        disabled={false}
        href={undefined}
        onClick={undefined}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchStart={[Function]}
        type="button"
      >
        <span
          aria-label=""
          className="material-icons"
          data-react-toolbox="font-icon"
        >
          menu
        </span>
      </button>
      <h1
        className=""
      >
        Hello World
      </h1>
      <nav
        className=""
        data-react-toolbox="navigation"
      >
        <a
          className=""
          data-react-toolbox="link"
          href="http://"
        >
          <span
            aria-label=""
            className="material-icons"
            data-react-toolbox="font-icon"
          >
            inbox
          </span>
          <abbr>
            Inbox
          </abbr>
        </a>
        <a
          className="undefined"
          data-react-toolbox="link"
          href="http://"
        >
          <span
            aria-label=""
            className="material-icons"
            data-react-toolbox="font-icon"
          >
            person
          </span>
          <abbr>
            Profile
          </abbr>
        </a>
      </nav>
    </div>
  </header>
  <div>
    <button
      className=""
      data-react-toolbox="button"
      disabled={false}
      href={undefined}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      Show Drawer
    </button>
  </div>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-6 col-md-3"
      >
        <div
          className="undefined"
          data-react-toolbox="card"
          style={
            Object {
              "width": "275px",
            }
          }
        >
          <div
            className=""
          >
            <div
              className=""
              data-react-toolbox="avatar"
            >
              <img
                alt=""
                className={undefined}
                src="https://placeimg.com/80/80/animals"
                title={undefined}
              />
            </div>
            <div>
              <h5
                className={undefined}
              >
                PG1: A
              </h5>
              <p
                className={undefined}
              >
                Programming Blogs
              </p>
            </div>
          </div>
          <div
            className="undefined"
            style={
              Object {
                "backgroundColor": undefined,
                "backgroundImage": "url('https://placeimg.com/800/450/nature')",
              }
            }
          >
            <div
              className=""
            />
          </div>
          <div
            className="undefined"
          >
            <div>
              <h5
                className={undefined}
              >
                Title goes here
              </h5>
              <p
                className={undefined}
              >
                Subtitle here
              </p>
            </div>
          </div>
          <div
            className=""
          >
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry
            </p>
          </div>
          <div
            className=""
          >
            <button
              className=""
              data-react-toolbox="button"
              disabled={false}
              href={undefined}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Action 1
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-xs-6 col-md-3"
      >
        <div
          className="undefined"
          data-react-toolbox="card"
          style={
            Object {
              "width": "275px",
            }
          }
        >
          <div
            className=""
          >
            <div
              className=""
              data-react-toolbox="avatar"
            >
              <img
                alt=""
                className={undefined}
                src="https://placeimg.com/80/80/animals"
                title={undefined}
              />
            </div>
            <div>
              <h5
                className={undefined}
              >
                PG1: B
              </h5>
              <p
                className={undefined}
              >
                Programming Blogs
              </p>
            </div>
          </div>
          <div
            className="undefined"
            style={
              Object {
                "backgroundColor": undefined,
                "backgroundImage": "url('https://placeimg.com/800/450/nature')",
              }
            }
          >
            <div
              className=""
            />
          </div>
          <div
            className="undefined"
          >
            <div>
              <h5
                className={undefined}
              >
                Title goes here
              </h5>
              <p
                className={undefined}
              >
                Subtitle here
              </p>
            </div>
          </div>
          <div
            className=""
          >
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry
            </p>
          </div>
          <div
            className=""
          >
            <button
              className=""
              data-react-toolbox="button"
              disabled={false}
              href={undefined}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Action 1
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-xs-6 col-md-3"
      >
        <div
          className="undefined"
          data-react-toolbox="card"
          style={
            Object {
              "width": "275px",
            }
          }
        >
          <div
            className=""
          >
            <div
              className=""
              data-react-toolbox="avatar"
            >
              <img
                alt=""
                className={undefined}
                src="https://placeimg.com/80/80/animals"
                title={undefined}
              />
            </div>
            <div>
              <h5
                className={undefined}
              >
                PG1: C
              </h5>
              <p
                className={undefined}
              >
                Programming Blogs
              </p>
            </div>
          </div>
          <div
            className="undefined"
            style={
              Object {
                "backgroundColor": undefined,
                "backgroundImage": "url('https://placeimg.com/800/450/nature')",
              }
            }
          >
            <div
              className=""
            />
          </div>
          <div
            className="undefined"
          >
            <div>
              <h5
                className={undefined}
              >
                Title goes here
              </h5>
              <p
                className={undefined}
              >
                Subtitle here
              </p>
            </div>
          </div>
          <div
            className=""
          >
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry
            </p>
          </div>
          <div
            className=""
          >
            <button
              className=""
              data-react-toolbox="button"
              disabled={false}
              href={undefined}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Action 1
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-xs-6 col-md-3"
      >
        <div
          className="undefined"
          data-react-toolbox="card"
          style={
            Object {
              "width": "275px",
            }
          }
        >
          <div
            className=""
          >
            <div
              className=""
              data-react-toolbox="avatar"
            >
              <img
                alt=""
                className={undefined}
                src="https://placeimg.com/80/80/animals"
                title={undefined}
              />
            </div>
            <div>
              <h5
                className={undefined}
              >
                Fight: Arrays
              </h5>
              <p
                className={undefined}
              >
                Code Fight Challenge
              </p>
            </div>
          </div>
          <div
            className="undefined"
            style={
              Object {
                "backgroundColor": undefined,
                "backgroundImage": "url('https://placeimg.com/800/450/nature')",
              }
            }
          >
            <div
              className=""
            />
          </div>
          <div
            className="undefined"
          >
            <div>
              <h5
                className={undefined}
              >
                Title goes here
              </h5>
              <p
                className={undefined}
              >
                Subtitle here
              </p>
            </div>
          </div>
          <div
            className=""
          >
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry
            </p>
          </div>
          <div
            className=""
          >
            <button
              className=""
              data-react-toolbox="button"
              disabled={false}
              href={undefined}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Action 1
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-xs-6 col-md-3"
      >
        <div
          className="undefined"
          data-react-toolbox="card"
          style={
            Object {
              "width": "275px",
            }
          }
        >
          <div
            className=""
          >
            <div
              className=""
              data-react-toolbox="avatar"
            >
              <img
                alt=""
                className={undefined}
                src="https://placeimg.com/80/80/animals"
                title={undefined}
              />
            </div>
            <div>
              <h5
                className={undefined}
              >
                Fight: Strings
              </h5>
              <p
                className={undefined}
              >
                Code Fight Challenge
              </p>
            </div>
          </div>
          <div
            className="undefined"
            style={
              Object {
                "backgroundColor": undefined,
                "backgroundImage": "url('https://placeimg.com/800/450/nature')",
              }
            }
          >
            <div
              className=""
            />
          </div>
          <div
            className="undefined"
          >
            <div>
              <h5
                className={undefined}
              >
                Title goes here
              </h5>
              <p
                className={undefined}
              >
                Subtitle here
              </p>
            </div>
          </div>
          <div
            className=""
          >
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry
            </p>
          </div>
          <div
            className=""
          >
            <button
              className=""
              data-react-toolbox="button"
              disabled={false}
              href={undefined}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Action 1
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-xs-6 col-md-3"
      >
        <div
          className="undefined"
          data-react-toolbox="card"
          style={
            Object {
              "width": "275px",
            }
          }
        >
          <div
            className=""
          >
            <div
              className=""
              data-react-toolbox="avatar"
            >
              <img
                alt=""
                className={undefined}
                src="https://placeimg.com/80/80/animals"
                title={undefined}
              />
            </div>
            <div>
              <h5
                className={undefined}
              >
                Day 1
              </h5>
              <p
                className={undefined}
              >
                Daily Summary
              </p>
            </div>
          </div>
          <div
            className="undefined"
            style={
              Object {
                "backgroundColor": undefined,
                "backgroundImage": "url('https://placeimg.com/800/450/nature')",
              }
            }
          >
            <div
              className=""
            />
          </div>
          <div
            className="undefined"
          >
            <div>
              <h5
                className={undefined}
              >
                Title goes here
              </h5>
              <p
                className={undefined}
              >
                Subtitle here
              </p>
            </div>
          </div>
          <div
            className=""
          >
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry
            </p>
          </div>
          <div
            className=""
          >
            <button
              className=""
              data-react-toolbox="button"
              disabled={false}
              href={undefined}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Action 1
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-xs-6 col-md-3"
      >
        <div
          className="undefined"
          data-react-toolbox="card"
          style={
            Object {
              "width": "275px",
            }
          }
        >
          <div
            className=""
          >
            <div
              className=""
              data-react-toolbox="avatar"
            >
              <img
                alt=""
                className={undefined}
                src="https://placeimg.com/80/80/animals"
                title={undefined}
              />
            </div>
            <div>
              <h5
                className={undefined}
              >
                Day 2
              </h5>
              <p
                className={undefined}
              >
                Daily Summary
              </p>
            </div>
          </div>
          <div
            className="undefined"
            style={
              Object {
                "backgroundColor": undefined,
                "backgroundImage": "url('https://placeimg.com/800/450/nature')",
              }
            }
          >
            <div
              className=""
            />
          </div>
          <div
            className="undefined"
          >
            <div>
              <h5
                className={undefined}
              >
                Title goes here
              </h5>
              <p
                className={undefined}
              >
                Subtitle here
              </p>
            </div>
          </div>
          <div
            className=""
          >
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry
            </p>
          </div>
          <div
            className=""
          >
            <button
              className=""
              data-react-toolbox="button"
              disabled={false}
              href={undefined}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Action 1
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-xs-6 col-md-3"
      >
        <div
          className="undefined"
          data-react-toolbox="card"
          style={
            Object {
              "width": "275px",
            }
          }
        >
          <div
            className=""
          >
            <div
              className=""
              data-react-toolbox="avatar"
            >
              <img
                alt=""
                className={undefined}
                src="https://placeimg.com/80/80/animals"
                title={undefined}
              />
            </div>
            <div>
              <h5
                className={undefined}
              >
                Day 3
              </h5>
              <p
                className={undefined}
              >
                Daily Summary
              </p>
            </div>
          </div>
          <div
            className="undefined"
            style={
              Object {
                "backgroundColor": undefined,
                "backgroundImage": "url('https://placeimg.com/800/450/nature')",
              }
            }
          >
            <div
              className=""
            />
          </div>
          <div
            className="undefined"
          >
            <div>
              <h5
                className={undefined}
              >
                Title goes here
              </h5>
              <p
                className={undefined}
              >
                Subtitle here
              </p>
            </div>
          </div>
          <div
            className=""
          >
            <p>
              Lorem Ipsum is simply dummy text of the printing and typesetting industry
            </p>
          </div>
          <div
            className=""
          >
            <button
              className=""
              data-react-toolbox="button"
              disabled={false}
              href={undefined}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Action 1
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
